Makefile.inc

```diff
@@ -119,12 +119,18 @@ $(OUTNAME).elf: $(OBJS) $(LIBOBJS) $(CORE_A)
 	"-Wl,-Map,$(OUTNAME).map" -o $@ "-L." "-L$(CORELIBS_DIR)/variants/arduino_101" \
 	-Wl,--whole-archive "-larc32drv_arduino101" -Wl,--no-whole-archive \
 	-Wl,--start-group "-larc32drv_arduino101" -lnsim -lc -lm -lgcc $^ 
-	cp $@ arc-debug.elf
 
-compile: $(OUTNAME).elf
-	$(OBJCOPY) -S -O binary -R .note -R .comment -R COMMON -R .eh_frame $< arc.bin
-	$(STRIP) $(OUTNAME).elf
-	$(OBJCOPY) -S -O ihex -R .note -R .comment -R COMMON -R .eh_frame  $< $(OUTNAME).hex
+do-compile: clean-all $(OUTNAME).elf
+	$(OBJCOPY) -S -O binary -R .note -R .comment -R COMMON -R .eh_frame $(OUTNAME).elf arc.bin
+	$(OBJCOPY) -S -O ihex -R .note -R .comment -R COMMON -R .eh_frame  $(OUTNAME).elf $(OUTNAME).hex
+
+compile: CFLAGS+=-Os
+compile: CPPFLAGS+=-Os
+compile: do-compile
+
+compile-debug: CFLAGS+=-g
+compile-debug: CPPFLAGS+=-g
+compile-debug: do-compile
 
 upload: compile
```

Comment:
`upload` target currently compiles before uploading. This was put in place to mimic how IDE re-compiles upon pressing upload. We should take it out.